### PR TITLE
NER Improvements and Guide

### PR DESCRIPTION
**Pull request by abidlabs**
This PR:
* makes it easier to do named-entity-recognition (NER) using HF `transformers` pipeline for NER by accepting the format that is returned from that pipeline for the `HighlightedText` component (closes: #1584) 
* also includes a Guide to explain how to do NER in more detail.

### CODE REVIEW

File: guides/named_entity_recognition.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+```py
+from transformers import pipeline 
+ner_pipeline = pipeline("ner")
+ner_pipeline("Does Chicago have any Pakistani restaurants")
+```
+
+Output:
+
+```
```
Comment (freddyaboulton) [inline guides/named_entity_recognition.md:47]: nit: Type this as "```bash" cause the format gets messed up during the rendering

image

File: guides/named_entity_recognition.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  'end': 12},
+ {'entity': 'I-MISC',
+  'score': 0.9958592,
+  'index': 5,
+  'word': 'Pakistani',
+  'start': 22,
+  'end': 31}]
+```
+
+If you have such a model, it is very easy to hook it up to Gradio's `HighlightedText` component. All you need to do is pass in this **list of entities**, along with the **original text** the the model, together as dictionary, with the keys being `"entities"` and `"text"` respectively.
```
Comment (freddyaboulton) [inline guides/named_entity_recognition.md:62]: nit: "the the model,"

File: gradio/components.py [hunk 3]
```diff
@@ -3235,6 +3235,22 @@ def postprocess(
         """
         if y is None:
             return None
+        if isinstance(y, dict):
```
Comment (freddyaboulton) [inline gradio/components.py:3238]: Implementation looks good to me but let's add some unit tests for this please. Some helpful cases would be no entities, all entities, example in the demo.

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @abidlabs ! Just left some nits.

**Comment by abidlabs** [comment]
Thanks @freddyaboulton, agreed with all feedback.

File: gradio/components.py (hunk 3 above)
Comment (abidlabs) [reply gradio/components.py:3238]: Done

**Comment by abidlabs** [comment]
All fixed up, going to merge it in

**Pull request merged**